### PR TITLE
[Remove] Joda from Serialization

### PR DESCRIPTION
**Pull request by nknize**
In [Legacy 7.0 joda time was removed](https://github.com/elastic/elasticsearch/pull/36363) in favor of java time to support nanoseconds properly. This PR removes joda from the serialization including ignoring the joda flag sent over the wire for DocValueFormat.

relates #9349 
depends on #9355

### CODE REVIEW

**Comment by nknize** [comment]
Opened #9355 as a prereq to this PR.

**Review by scrawfor99 — APPROVED** [review]

File: server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java [hunk 1]
```diff
@@ -141,14 +127,11 @@ public Map<Class<?>, XContentBuilder.HumanReadableTransformer> getXContentHumanR
...
         Map<Class<?>, Function<Object, Object>> transformers = new HashMap<>();
-        transformers.put(Date.class, d -> DEFAULT_DATE_PRINTER.print(((Date) d).getTime()));
-        transformers.put(DateTime.class, d -> DEFAULT_DATE_PRINTER.print((DateTime) d));
-        transformers.put(MutableDateTime.class, d -> DEFAULT_DATE_PRINTER.print((MutableDateTime) d));
-        transformers.put(ReadableInstant.class, d -> DEFAULT_DATE_PRINTER.print((ReadableInstant) d));
-        transformers.put(Long.class, d -> DEFAULT_DATE_PRINTER.print((long) d));
-        transformers.put(Calendar.class, d -> DEFAULT_DATE_PRINTER.print(((Calendar) d).getTimeInMillis()));
-        transformers.put(GregorianCalendar.class, d -> DEFAULT_DATE_PRINTER.print(((Calendar) d).getTimeInMillis()));
-        transformers.put(Instant.class, d -> DEFAULT_DATE_PRINTER.print((Instant) d));
+        transformers.put(Date.class, d -> DEFAULT_FORMATTER.format(((Date) d).toInstant()));
```
[the remark this answers is not in the document]
Comment (reta) [reply server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java:130]: No no, only for these ones, fe:

`
       transformers.put(Long.class, d -> DateFormatter.ISO_8601.format(Instant.ofEpochMilli((long) d)));
        transformers.put(Calendar.class, d -> DateFormatter.ISO_8601.format(((Calendar) d).toInstant()));
        transformers.put(GregorianCalendar.class, d -> DateFormatter.ISO_8601.format(((Calendar) d).toInstant()));
        transformers.put(Instant.class, d -> DateFormatter.ISO_8601.format((Instant) d));
`

[the remark this answers is not in the document]
Comment (nknize) [reply server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java:130]: I like that idea.

[the remark this answers is not in the document]
Comment (nknize) [reply server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java:130]: @reta I'm concerned about `ISO_8601` not being compatible with the spec. Probably safer to stick with `DEFAULT_FORMATTER` here and explore changing later?

[the remark this answers is not in the document]
Comment (reta) [reply server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java:130]: @nknize I did a quick test and DEFAULT_FORMATTER / DateFormatter.ISO_8601 / DEFAULT_DATE_PRINTER produce the same results (the DateFormatter.ISO_8601 is backed by "strict_date_optional_time_nanos" formatter), we should be all good here

[the remark this answers is not in the document]
Comment (nknize) [reply server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java:130]: Excellent! Thx for verifying. Let me know if you're good with this and I'll get it merged and backported.

**Comment by reta** [comment]
@nknize LGMT but could we now drop `api "joda-time:joda-time:${versions.joda}"` from server deps?

**Comment by nknize** [comment]
> LGMT but could we now drop `api "joda-time:joda-time:${versions.joda}"` from server deps?

@reta Not yet. `DateObjectValuesSource` and `DateObject` still need to be switched to java time. I've got a follow up PR to take care of that. I'm just trying to keep these PRs from getting super large unless they have to.

**Review by reta — APPROVED** [review]

**Pull request merged**